server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -505,16 +505,17 @@ public ActionModule(
             actionPlugins.stream().flatMap(p -> p.getRestHeaders().stream()),
             Stream.of(new RestHeaderDefinition(Task.X_OPAQUE_ID, false))
         ).collect(Collectors.toSet());
-        UnaryOperator<RestHandler> restWrapper = null;
+        List<UnaryOperator<RestHandler>> restWrappers = new ArrayList<>();
         // Only one plugin is allowed to have a rest wrapper. i.e. Security plugin
         for (ActionPlugin plugin : actionPlugins) {
             UnaryOperator<RestHandler> newRestWrapper = plugin.getRestHandlerWrapper(threadPool.getThreadContext());
             if (newRestWrapper != null) {
                 logger.debug("Using REST wrapper from plugin " + plugin.getClass().getName());
-                if (restWrapper != null) {
-                    throw new IllegalArgumentException("Cannot have more than one plugin implementing a REST wrapper");
+                restWrappers.add(newRestWrapper);
+                // Identity Module + one other plugin
+                if (restWrappers.size() > 2) {
```

Comment:
Now that we have two handlerWrappers, what do you think about adding additional cases to any tests involving the original to make sure that the secondary wrapper for the Identity module performs as expected? 

Likewise, is there anything that we would need to add further testing for because of using two wrappers? I am not sure but perhaps you would have some idea given your time implementing the second wrapper.

Reply:
Absolutely, we will need to expand with many more scenarios of having the two systems running together. A basic test case we will need to do is make sure the entire suite of tests in the security plugin work when the identity module exists, but is disabled.

We need to think through scenarios of where the systems can work together. I'm not entirely sure how it should work. 

1 scenario: Identity enable + security plugin installed and enabled. user1 exists in identity and doesn't exist in security plugin internal user db. If the rest request is authenticated and authorized with identity can it skip the rest handler wrapper of the security plugin? 

Security plugin has a notion of order when it comes to authentication backends, will we also need to introduce order here to be able to configure which rest handler wrapper is applied first?

Reply:
Gotcha. Thank you for the detailed explanation. I am glad to see you already have some ideas for the possible cases that could be added to support this with testing. Let me know if you need a hand with these.

Reply:
I think this is OK to merge to keep moving fast and before we invest in supporting a list of handlers in main we should double check the pros/cons of this design.

IMO I think Identity module can use existing plugin interfaces but I'm not sure this scenario is a good fit.  Explicitly calling AuthenticationManager in the RestHandler is non-ambiguous when and how this authentication is transmitted.